### ISSUE TITLE
disableAsyncTimeout issue

### ISSUE THREAD

**Issue by melamriD365** [MicrosoftDocs/powerapps-docs#2978]
The new "disableAsyncTimeout" feature does not work correctly in my environment. Indeed, the timeout is still valid, the error message below appears after 10 seconds.

<img width="281" alt="image" src="https://user-images.githubusercontent.com/36016227/149290979-84266991-4080-45b2-8865-92513e6e38d0.png">

Environment details: 
2021 release wave 2 enabled
Server version: 9.2.21121.00121
Client version: 1.4.3664-2112.1

Thanks !

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 3c3168e3-774b-aad1-0cd2-d7e6eb487088
* Version Independent ID: 2fb2edc5-b09d-bfaa-0644-675d29e00598
* Content: [Form OnSave event (Client API reference) in model-driven apps - Power Apps](https://docs.microsoft.com/en-us/powerapps/developer/model-driven-apps/clientapi/reference/events/form-onsave)
* Content Source: [powerapps-docs/developer/model-driven-apps/clientapi/reference/events/form-onsave.md](https://github.com/MicrosoftDocs/powerapps-docs/blob/main/powerapps-docs/developer/model-driven-apps/clientapi/reference/events/form-onsave.md)
* Service: **powerapps**
* GitHub Login: @Nkrb
* Microsoft Alias: **nabuthuk**

**Comment by melamriD365** [comment]
Same issue in this environment: 

Server version: 9.2.21123.00167
Client version: 1.4.3726-2112.3

Thanks !

**Comment by melamriD365** [comment]
Same issue in this environment:

Server version: 9.2.22015.00127
Client version: 1.4.3763-2201.5

Thanks !

**Comment by hsvensson1976** [comment]
I have the same issue. Calling disableAsyncTimeout makes no difference at all.

Server version: 9.2.22034.00151
Client version: 1.4.4090-2203.4

**Comment by melamriD365** [comment]
Same issue in this environment:

Server version: 9.2.22054.000139
Client version: 1.4.4277-2205.4

Could you please give us a feedback?

Thanks, 
Mehdi

**Comment by hsvensson1976** [comment]
I opened an MS ticket on this and I was informed about a work-around. If you add the onSave event via the form designer (i.e. not by code) and include the line: 

executionContext.getEventArgs().disableAsyncTimeout();

(given your execution context input parameter of your save function is called executionContext)

…at the top of your onSave function you may still get a timeout error in the console log, but it will not result in a pop-up error message being displayed to the user. This was enough to make things work in my case. 

Also, if I recall correctly, there is another thing to be aware of, and that is that whether the onLoad event is executed after a save does not seem to be consistent between the two ways of adding an onSave event (form designer/code), so changing from one to the other may have other unwanted consequences.

**Comment by scoobaroo** [comment]
I don't think not showing a pop-up error to the user is good enough for our scenario. We are doing some address verification via 3rd party API, and in instances where it's not verified, we pop up a dialog. Depending on whether or not the user confirms or not, we will either continue with the save or prevent the save. (This is on a quick create form.) With a timeout error, I don't think the code afterwards will execute. I appreciate your insight though. Another question: I heard that you can't prevent save in an async on Save function. If this is not possible, what kind of workaround do we have?

**Comment by melamriD365** [comment]
For me, this is a product bug. We must have the possibility to register the handler dynamically. This bug has been there since November 2021. I have done my best to get an answer...Without result.

**Comment by hsvensson1976** [comment]
Yes, absolutely a bug. Which is what MS also confirmed to me in a response within my support ticket. 

Just wanted to mention the work-around, since it was useful in at least my scenario.

**Comment by scoobaroo** [comment]
@hsvensson1976 Thank you so much for your comment. What you say is 100% correct. Moving it to the form properties rather than programmatically fixed our issue. @melamriD365  is correct though, we should be able to register the handler dynamically.
Thanks so much for the tips!
